conceptual/Npgsql/security.md

```diff
@@ -10,18 +10,63 @@ If a `Password` is not specified and your PostgreSQL is configured to request a
+Starting with 6.0, the following `SSL Mode` values are supported (see the [PostgreSQL docs](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-SSLMODE-STATEMENTS) for more details):
 
-## Encryption (SSL/TLS)
+SSL Mode    | Eavesdropping protection | Man-in-the-middle protection | Statement
+----------- | ------------------------ | ---------------------------- | ---------
+Disable     | No                       | No                           | I don't care about security, and I don't want to pay the overhead of encryption.
+Allow       | Maybe                    | No                           | I don't care about security, but I will pay the overhead of encryption if the server insists on it.
+Prefer      | Maybe                    | No                           | I don't care about encryption, but I wish to pay the overhead of encryption if the server supports it.
+Require*    | Yes                      | No                           | I want my data to be encrypted, and I accept the overhead. I trust that the network will make sure I always connect to the server I want.
+VerifyCA    | Yes                      | Depends on CA policy         | I want my data encrypted, and I accept the overhead. I want to be sure that I connect to a server that I trust.
+VerifyFull  | Yes                      | Yes                          | I want my data encrypted, and I accept the overhead. I want to be sure that I connect to a server I trust, and that it's the one I specify.
+
+* `SSL Mode=Require` currently requires explicitly setting `Trust Server Certificate=true` as well; this requirement will be removed in a future version. This combination should be used with e.g. self-signed certificates.
+
+The default mode in 6.0 is `Prefer`, which allows SSL but does not require it, and does not validate certificates.
+
+### Older versions
+
+Versions prior to 6.0 supported the following `SSL Mode` values:
+
+SSL Mode    | Eavesdropping protection | Man-in-the-middle protection | Statement
+----------- | ------------------------ | ---------------------------- | ---------
+Disable     | No                       | No                           | I don't care about security, and I don't want to pay the overhead of encryption.
+Prefer      | Maybe                    | No                           | I don't care about encryption, but I wish to pay the overhead of encryption if the server supports it.
+Require     | Yes                      | No                           | I want my data encrypted, and I accept the overhead. I want to be sure that I connect to a server I trust, and that it's the one I specify.
+
+The default mode prior was 6.0 is `Disable`.
+
+To disable certificate validation when using `Require`, set `Trust Server Certificate` to true; this allows connecting to servers with e.g. self-signed certificates.
```

Comment:
That's also true for `Prefer`, so we can mention it here.

Reply:
True, though that doesn't ensure encryption is being used. I added "while still requiring encryption", let me know if that seems sufficient.